google/cloud/spanner/connection.h

```diff
@@ -128,6 +128,9 @@ class Connection {
   };
   //@}
 
+  /// Returns any options use by the concrete Connection.
+  virtual Options options() { return Options{}; }
```

Comment:
This is a new public API, a few questions:

- Is this something we want to expose to customers?  If not, maybe we should consider a `private` + `friend in internal` idiom?
- If it is a public API we want to expose, is this something that we should add to the `MockConnection` class?  Would customers ever want to use it?

Reply:
Added to `MockConnection`.

Reply:
Ack.   Let's talk about this on the team meeting.  You should merge, we can always change this before the release, or something.  My experience with exposing a `connection_options()` in `storage` was not great.  With that said, this new function returns by value, and the default return is sensible for tests, so many of my concerns do not apply.  I do worry about adding APIs, maybe too much.

Reply:
Ack.